File: lib_common/src/main/java/com/wujia/lib_common/data/network/DownloadCallBack.java
package com.wujia.lib_common.data.network;

public abstract class DownloadCallBack {
    public void onStart() {
    }

    public void onCompleted() {
    }

    abstract public void onError(Throwable e);

    public void onProgress(long fileSizeDownloaded) {
    }

    abstract public void onSucess(String path, long fileSize);
}

File: lib_common/src/main/java/com/wujia/lib_common/data/network/DownloadCallBack.kt
package com.wujia.lib_common.data.network

abstract class DownloadCallBack {
    fun onStart() {}

    fun onCompleted() {}

    abstract fun onError(e: Throwable)

    fun onProgress(fileSizeDownloaded: Long) {}

    abstract fun onSucess(path: String, fileSize: Long)
}
